.github/workflows/_Test-OCI-Factory.yaml

```diff
@@ -15,7 +15,58 @@ on:
       - "!src/workflow-engine/**"
       - "!src/cli-client/**"
 
+env:
+  # local path to clone the oci-factory to
+  OCI_FACTORY_DIR: oci-factory/
+
+  # path of pytest junit output
+  PYTEST_RESULT_PATH: pytest_results.xml
+
+
 jobs:
+
+  pytest:
+    # Trigger python unit tests across the repository
+    name: pytest
+    runs-on: ubuntu-22.04
+    steps:
+
+      # Job Setup
+      - uses: actions/checkout@v4
+        with:
+          path: ${{ env.OCI_FACTORY_DIR }}
+          fetch-depth: 1
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+      
+      # Note: Add additional dependency installation lines as required below 
+      # test-oci-factory/pytest requirements
+      - run: pip install -r ${{ env.OCI_FACTORY_DIR }}/src/test-oci-factory/pytest/requirements.txt
+
+
+      - name: Run pytest
+        continue-on-error: true
+        run: |
+          python3 -m pytest --junit-xml "${{ env.PYTEST_RESULT_PATH }}" "${{ env.OCI_FACTORY_DIR }}"
+
+      - name: Generate Summary
+        if: always()
```

Comment:
We can use ${{ !cancelled() }} instead since the workflow can be interrupted at previous steps --- A suggestion from the [Github Documentation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#always).

Reply:
Nice, lets do that